src/extensions/surveys/action-matcher.ts

```diff
@@ -0,0 +1,258 @@
+            this.actionRegistry?.add(action)
+            action.steps?.forEach((step) => {
+                this.actionEvents?.add(step?.event || '')
+            })
+        })
+        // eslint-disable-next-line no-console
+        console.log(`in action-matcher eventNames known are `, this.actionEvents.values())
+
+        if (this.instance?.autocapture) {
+            const selectorsToWatch: Set<string> = new Set<string>()
+            actions.forEach((action) => {
+                action.steps?.forEach((step) => {
+                    if (step?.selector) {
+                        selectorsToWatch.add(step?.selector)
+                    }
+                })
+            })
+            this.instance?.autocapture.setElementSelectors(selectorsToWatch)
+        }
+
+        // if (eventNames.length > 0) {
+        //     throw new Error(`I know about these events : ${eventNames}`)
+        // }
+    }
+
+    on(eventName: string, eventPayload?: CaptureResult) {
+        if (eventPayload == null || eventName.length == 0) {
+            return
+        }
+
+        if (!this.actionEvents.has(eventName) && !this.actionEvents.has(<string>eventPayload?.event)) {
+            // throw new Error(`unknown event ${eventName}, I only know about : ${JSON.stringify(this.actionEvents.values())}`)
+            return
+        }
+
+        if (this.actionRegistry && this.actionRegistry?.size > 0) {
+            this.actionRegistry.forEach((action) => {
+                if (this.checkAction(eventPayload, action)) {
+                    // // eslint-disable-next-line no-console
+                    // console.log(`in action-matcher emitting observed event for action`, action.name)
+                    this._debugEventEmitter.emit('actionCaptured', action.name)
+                }
+            })
+        }
+    }
+
+    _addActionHook(callback: (actionName: string, eventPayload?: any) => void): void {
+        this.onAction('actionCaptured', (data) => callback(data))
+    }
+
+    private checkAction(event?: CaptureResult, action?: ActionType): boolean {
+        if (action?.steps == null) {
+            return false
+        }
+
+        for (const step of action.steps) {
+            if (this.checkStep(event, step)) {
+                return true
+            }
+        }
+
+        return false
+    }
+
+    onAction(event: 'actionCaptured', cb: (...args: any[]) => void): () => void {
+        return this._debugEventEmitter.on(event, cb)
+    }
+
+    private checkStep = (event?: CaptureResult, step?: ActionStepType): boolean => {
+        // eslint-disable-next-line no-console
+        console.log(
+            ` in checkStep,  event is [${event?.event}],  step.event is ${
+                step?.event
+            } checkStepEvent is [${this.checkStepEvent(event, step)}] checkStepUrl  is [${this.checkStepUrl(
+                event,
+                step
+            )}] checkStepElement is [${this.checkStepElement(event, step)}]`
+        )
```

Comment:
I assume you didn't want to leave any of these console.logs behind :)

I don't use eslint-disable to make sure the pre-commit check catches it.

```suggestion
```